tfe/resource_tfe_team_members.go

```diff
@@ -0,0 +1,161 @@
+	if err != nil {
+		return fmt.Errorf("Error adding users to team %s: %v", teamID, err)
+	}
+
+	d.SetId(teamID)
+
+	return nil
+}
+
+func resourceTFETeamMembersRead(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Get the team ID and username..
+	teamID := d.Get("team_id").(string)
+
+	log.Printf("[DEBUG] Read users from team: %s", teamID)
+	users, err := tfeClient.TeamMembers.List(ctx, teamID)
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			log.Printf("[DEBUG] Users do no longer exist")
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading users from team %s: %v", teamID, err)
+	}
+
+	usernames := resourceTFETeamMembers().Schema["usernames"].ZeroValue().(*schema.Set)
+	for _, user := range users {
+		usernames.Add(user.Username)
+	}
+
+	if usernames.Len() > 0 {
+		d.Set("usernames", usernames)
+	} else {
+		log.Printf("[DEBUG] Users do no longer exist")
+		d.SetId("")
+	}
+
+	return nil
+}
+
+func resourceTFETeamMembersUpdate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	if d.HasChange("usernames") {
+		old, new := d.GetChange("usernames")
+		oldUsers := old.(*schema.Set).Difference(new.(*schema.Set))
+		newUsers := new.(*schema.Set).Difference(old.(*schema.Set))
+
+		// Get the team ID and username..
+		teamID := d.Get("team_id").(string)
+
+		// First add the new users.
+		if newUsers.Len() > 0 {
+			// Create a new options struct.
+			options := tfe.TeamMemberAddOptions{}
+
+			// Add all the users that need to be added.
+			for _, username := range newUsers.List() {
+				options.Usernames = append(options.Usernames, username.(string))
+			}
+
+			log.Printf("[DEBUG] Add users to team: %s", teamID)
+			err := tfeClient.TeamMembers.Add(ctx, teamID, options)
+			if err != nil {
+				return fmt.Errorf("Error adding users to team %s: %v", teamID, err)
+			}
+		}
+
+		// Then delete all the old users.
+		if oldUsers.Len() > 0 {
+			// Create a new options struct.
+			options := tfe.TeamMemberRemoveOptions{}
+
+			// Add all the users that need to be added.
+			for _, username := range oldUsers.List() {
+				options.Usernames = append(options.Usernames, username.(string))
+			}
+
+			log.Printf("[DEBUG] Remove users from team: %s", teamID)
+			err := tfeClient.TeamMembers.Remove(ctx, teamID, options)
+			if err != nil {
+				return fmt.Errorf("Error removing users to team %s: %v", teamID, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceTFETeamMembersDelete(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Get the team ID and username..
+	teamID := d.Get("team_id").(string)
+
+	// Create a new options struct.
+	options := tfe.TeamMemberRemoveOptions{}
+
+	// Add all the users that need to be added.
+	for _, username := range d.Get("usernames").(*schema.Set).List() {
+		options.Usernames = append(options.Usernames, username.(string))
+	}
```

Comment:
this can be removed (also d.Get("usernames") should return nil/an empty set here when the config no longer exists)

Reply:
Why can this be removed? Ow because if the config is deleted this is now and empty set... Check so the solution is that I query the API for all the members of the team and then delete those.

Fixed in the next commit.